src/ansi-c/c_typecheck_initializer.cpp

```diff
@@ -1036,9 +1036,9 @@ exprt c_typecheck_baset::do_initializer_list(
      to_array_type(full_type).size().is_nil())
   {
     // make complete by setting array size
-    size_t size=result.operands().size();
-    result.type().id(ID_array);
-    result.type().set(ID_size, from_integer(size, c_index_type()));
+    array_typet &array_type = to_array_type(result.type());
```

Comment:
`result.type()` is nil at this point.

Reply:
No, it was set to whatever type the initializer had.